### ISSUE TITLE
Incompatibility with Python 3.12.0b1

### ISSUE THREAD

**Issue by freakboy3742** [beeware/rubicon-objc#313]
### Describe the bug

Python 3.12.0b1 appears to have introduced something that breaks Rubicon; `from rubicon.objc import NSObject` raises a TypeError.

### Steps to reproduce

1. `from rubicon.objc import NSObject`
2. See error
```
Traceback (most recent call last):
  File "/Users/rkm/beeware/rubicon/objc/testbed.py", line 3, in <module>
    from rubicon.objc import NSObject, objc_method
  File "/Users/rkm/beeware/rubicon/objc/src/rubicon/objc/__init__.py", line 33, in <module>
    from . import api, collections, runtime, types
  File "/Users/rkm/beeware/rubicon/objc/src/rubicon/objc/api.py", line 1694, in <module>
    NSNumber = ObjCClass("NSNumber")
               ^^^^^^^^^^^^^^^^^^^^^
  File "/Users/rkm/beeware/rubicon/objc/src/rubicon/objc/api.py", line 1402, in __new__
    self = super().__new__(cls, ptr, objc_class_name, (ObjCInstance,), new_attrs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/rkm/beeware/rubicon/objc/src/rubicon/objc/api.py", line 927, in __new__
    super(ObjCInstance, type(self)).__setattr__(self, "ptr", object_ptr)
TypeError:  expected 2 arguments, got 3
```

### Expected behavior

The import should succeed.

### Environment

- Operating System: macOS Ventura 13.3.1
- Python version: 3.12.0b1
- Software versions:
  - Rubicon-Objc: 35b6fc8fb40043d51dd7e41d359cf8ef28932e8b 

### Additional context

There was no problem with 3.12.0a7; I haven't narrowed down the change that has caused the problem, and the Python 3.12 release notes don't have anything that is an obvious candidate (to me, anyway).

**Comment by freakboy3742** [comment]
I've narrowed the cause down to python/cpython#87729. Trying to work out a reproduction case that triggered the problem.

**Comment by freakboy3742** [comment]
Reproduction case found, and logged as python/cpython#105035.

**Comment by freakboy3742** [comment]
This has been fixed upstream, and should be included in 3.12.0b2.

**Comment by freakboy3742** [comment]
3.12.0b2 and b3 have been released, and CI builds indicate this issue has been resolved.
